packages/vite/src/node/packages.ts

```diff
@@ -125,17 +130,15 @@ export function findNearestPackageData(
     }
 
     const pkgPath = path.join(basedir, 'package.json')
-    try {
-      if (fs.statSync(pkgPath, { throwIfNoEntry: false })?.isFile()) {
-        const pkgData = loadPackageData(pkgPath)
-
-        if (packageCache) {
-          setFnpdCache(packageCache, pkgData, basedir, originalBasedir)
-        }
+    if (tryStatSync(pkgPath)?.isFile()) {
+      const pkgData = loadPackageData(pkgPath)
```

Comment:
We still have to wrap this `loadPackageData` with a try-catch`. 
This function can throw an error if the content is not readable (e.g. lack of permission) or the content is not a valid JSON.
https://github.com/vitejs/vite/blob/fdd466946c2a7fe4ff1a6d1b5950820b310c61f9/packages/vite/src/node/packages.ts#L166